File: 1642-water-bottles/1642-water-bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int sum = numBottles;

    while(numBottles>=numExchange){
        int rem= numBottles%numExchange;
        numBottles = numBottles/numExchange;
        sum += numBottles;
        numBottles = numBottles + (rem);
    }

    return sum;
    }
}

File: 1642-water-bottles/1642-water-bottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function(numBottles, numExchange) {

    let sum = numBottles;

    while(numBottles>=numExchange){
        let rem= numBottles%numExchange
        numBottles = Math.floor(numBottles/numExchange);
        sum += numBottles;
        console.log(numBottles%numExchange)
        numBottles = numBottles + (rem);
        console.log(numBottles, sum)
    }

    return sum;
};